### PR TITLE
Mathml evaluation

### LINKED ISSUE #97: Do the test_mathml_evaluation protocol
Part of #23.

Following from #95, needs support for:
- [x] Non oxmeta annotations - might already work?
- [x] More mathematics in `define` clauses, i.e. write lots of `to_sympy` methods.

### PR DESCRIPTION
**Pull request by MichaelClerx**
Fixes #97 

Seems to pass now, but doesn't have trig functions and raises 2 warnings

### CODE REVIEW

**Comment by MichaelClerx** [comment]
Ready for review now @jonc125

File: fc/parsing/actions.py [hunk 1]
```diff
@@ -1119,6 +1238,34 @@ def _add_input_variables(self):
...
+            except KeyError:
+
+                # Find variable in input or output declarations
+                try:
+                    var = variables[eq.var.name()]
+                except KeyError:
+                    continue
+
+                # Add the new variable, with ontology annotation
+                name = self.model.get_unique_name('protocol__' + var.local_name)
```
Comment (jonc125) [inline fc/parsing/actions.py]: Now these 5 lines are in two places, I'd extract a helper method!

File: fc/templates/weblab_model.pyx [hunk 2]
```diff
@@ -227,6 +224,9 @@ cdef class {{ class_name }}(CvodeSolver):
             'rdf': self.env,
             'cellml': self.env,
             'vCard': self.env,
+            {%- for prefix in ns_map %}
```
Comment (jonc125) [inline fc/templates/weblab_model.pyx]: You can get rid of everything apart from this loop now.

File: test/test_mathml_evaluation.py [hunk 3]
```diff
@@ -1,22 +1,18 @@
+"""Test implementation of voltage clamps.
```
Comment (jonc125) [inline test/test_mathml_evaluation.py]: I think some dodgy copy & paste has happened here in the past!

Comment (MichaelClerx) [reply test/test_mathml_evaluation.py]: I copied the docstring from the class to the module level. Turned it into comments now if you prefer?

Comment (jonc125) [reply test/test_mathml_evaluation.py]: It wasn't that, it's that the docstring doesn't match what the test is doing :)

Comment (MichaelClerx) [reply test/test_mathml_evaluation.py]: Fair enough. I forgot to read :man_facepalming:

**Pull request merged**